ably-ios/ARTDataEncoder.m

```diff
@@ -0,0 +1,175 @@
+//
+//  ARTDataEncoder.m
+//  ably-ios
+//
+//  Created by Jason Choy on 18/12/2014.
+//  Copyright (c) 2014 Ably. All rights reserved.
+//
+
+#import "ARTCrypto.h"
+#import "ARTLog.h"
+#import "ARTDataEncoder.h"
+
+@interface ARTDataEncoder ()
+
+ART_ASSUME_NONNULL_BEGIN
+
+@property (readonly, nonatomic) ARTLog *logger;
+
+ART_ASSUME_NONNULL_END
+
+@end
+
+@implementation ARTDataEncoder {
+    id<ARTChannelCipher> _cipher;
+    ARTLog *_logger;
```

Comment:
You don't need this. You already have the property `logger` which does autosynthesis and you can access the instance variable directly with `_logger`.